### PR TITLE
Work on Qemu Log Viewer

### PR DESCRIPTION
**Pull request by laerreal**
Code refactoring

### v4
* organize style tags for trace text

### v3
* eliminate `insert_diff` and related code
* fix "usege"

### v2
* CPU state difference highlighting is removed (for a next merge)

### CODE REVIEW

File: qlv.py [hunk 1]
```diff
@@ -71,6 +71,211 @@ def __init__(self, master, **kw):
...
+            sashrelief = RAISED
+        )
+        panes.add(panes_trace_text)
+
+        self.qlog_trace_texts = []
+
+    def show_logs(self, qlogs):
+        panes_trace_text = self.panes_trace_text
+        qlog_trace_texts = self.qlog_trace_texts
+        # TODO: re-usege?
```
Comment (BDanAnd) [inline qlv.py]: re-usage

File: qlv.py [hunk 2]
```diff
@@ -71,6 +71,211 @@ def __init__(self, master, **kw):
...
+                if qlog_idx == 0:
+                    left_trace = i.trace.as_text
+                    trace_text.insert(END, left_trace)
+                else:
+                    cur_trace = i.trace.as_text
+                    if left_trace is None:
+                        # Nothing to diff
+                        trace_text.insert(END, cur_trace)
+                    else:
+                        insert_diff(trace_text, left_trace, cur_trace)
```
Comment (BDanAnd) [inline qlv.py]: `insert_diff` doesn't exist in this MR now.

**Comment by laerreal** [comment]
v3 is ready!

File: qlv.py [hunk 3]
```diff
@@ -71,6 +71,200 @@ def __init__(self, master, **kw):
...
+
+            trace_text.grid(row = 0, column = 0, sticky = "NESW")
+
+            add_scrollbars_native(fr_trace_text, trace_text)
+
+            trace_text.tag_configure("file", foreground = "#AAAAAA")
+            trace_text.tag_configure("warning", foreground = "#FFBB66")
+            trace_text.tag_configure(STYLE_DIFFERENCE[0],
+                foreground = "#FF0000"
+            )
```
Comment (BDanAnd) [inline qlv.py]: This tag not used for text now.

File: qlv.py [hunk 4]
```diff
@@ -71,6 +71,200 @@ def __init__(self, master, **kw):
...
+
+            trace_text = GUIText(fr_trace_text, state = READONLY, wrap = NONE)
+            qlog_trace_texts.append(trace_text)
+
+            trace_text.grid(row = 0, column = 0, sticky = "NESW")
+
+            add_scrollbars_native(fr_trace_text, trace_text)
+
+            trace_text.tag_configure("file", foreground = "#AAAAAA")
+            trace_text.tag_configure("warning", foreground = "#FFBB66")
```
Comment (BDanAnd) [inline qlv.py]: Maybe use `STYLE_FILE[0]` and `STYLE_WARNING[0]`?

**Comment by laerreal** [comment]
v4 is ready

**Pull request merged**